Set hideResult flag for cse variables

rfranke opened the pull request:
It was already set in function createCSEVar, but not in createCSEArrayVar.

### Related Issues

#8009

### Purpose

Presentation of results should not depend on naming conventions (starting with $ or _D_).

### Approach

Set hideResult flag, see also variables created for when conditions.

rfranke committed:
Set hideResult flag for cse variables

It was already set in function createCSEVar, but not in createCSEArrayVar.

lochel reviewed (commented):
This looks good to me. Any idea why the test suite fails?

rfranke committed:
Adapt optimization test runExReduceDrumBoiler.mos to emit start values

rfranke committed:
Don't use index counter as hideResult flag

This looks like a copy/paste error from SIMVAR to VAR that has
not been detected by compiler type checks.
